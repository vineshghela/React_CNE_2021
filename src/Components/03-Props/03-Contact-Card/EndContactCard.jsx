

import EndContactCardBluePrint from "./EndContactCardBluePrint"

const EndContactCard =()=>{
    return (
        <div className="Contact-card-list">
          <EndContactCardBluePrint
            whatPet="This is a dog"
            petName="Harry Pawter"
            phone="(+44)77777777"
            email="jimmy@chew.pup"
            img="https://placedog.net/samples/2.jpg"
          />
          <EndContactCardBluePrint
            petName="jimmy chew"
            phone="(+44)77777777"
            email="jimmy@chew.pup"
            img="https://placedog.net/samples/5.jpg"
          />
          <EndContactCardBluePrint
            petName="Bark Wahlberg"
            phone="(+44)77777777"
            email="BarkWahlberg@chew.pup"
            img="https://placedog.net/samples/15.jpg"
          />
          <EndContactCardBluePrint
            petName="Indiana Bones"
            phone="(+44)77777777"
            email="IndianaBones@chew.pup"
            img="https://placedog.net/samples/5.jpg"
          />
        </div>
    )

}

export default EndContactCard;