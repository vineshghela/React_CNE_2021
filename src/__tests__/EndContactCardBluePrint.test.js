import {create} from 'react-test-renderer';
import '@testing-library/jest-dom';
import {expect, it} from '@jest/globals';
import EndContactContactCardBluePrint from '../Components/03-Props/03-Contact-Card/EndContactCardBluePrint';
import EndContactCard from '../Components/03-Props/03-Contact-Card/EndContactCard ';

describe(`End contact card blue print testing with props`,()=>{

    const props={
        whatPet:"This is a dog",
        petName:"Harry Pawter",
        phone:"(+44)77777777",
        email:"jimmy@chew.pup",
        img:"https://placedog.net/samples/2.jpg"
    }
    
    let componentToTest;

    beforeEach(()=>{
        const testInstance = create(<EndContactContactCardBluePrint{...props}/>);
        componentToTest = testInstance.root;
        // console.log("Before each",componentToTest);
    })

    it(`should render a H3 with the name of the pet`,()=>{
        const h3 = componentToTest.findByType(`h3`);
        // console.log("my h3",h3);
        expect(h3.children).toEqual([props.petName]);
    })
    it(`should render phone number in the first p tag`,()=>{
        const firstP = componentToTest.findAllByType(`p`)[0];
        console.log(firstP.children[1]);
    })
    it(`should render email for the second p tag`,()=>{
        const secondP = componentToTest.findAllByType(`p`)[1];
        console.log(secondP.children[1]);
    })

    it(`should match the snapshot`,()=>{
        const instance = create(<EndContactContactCardBluePrint {...props}/>).toJSON();
        expect(instance).toMatchSnapshot();
    })

    const mockChildComponent = jest.mock(`../Components/03-Props/03-Contact-Card/EndContactCardBluePrint`,()=>{
        return EndContactCard=(props)=>{
            return <p>Mocked Component</p>;
        }
    })

    console.log(mockChildComponent);

    it(`it should match the snapshot`,()=>{
        const instance = create(<mockChildComponent/>).toJSON();
        expect(instance).toReturnWith(mockChildComponent);
    })
})
