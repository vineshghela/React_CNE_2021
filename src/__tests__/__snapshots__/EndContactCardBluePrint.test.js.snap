// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End contact card blue print testing with props should match the snapshot 1`] = `
<div
  className="contact-card"
>
  <h2>
    This is a dog
  </h2>
  <img
    alt="this is a of a dog"
    src="https://placedog.net/samples/2.jpg"
  />
  <h3>
    Harry Pawter
  </h3>
  <p>
    Phone: 
    (+44)77777777
  </p>
  <p>
    Email:
    jimmy@chew.pup
  </p>
</div>
`;
